### PR TITLE
Feat/add cart item

### PR DESCRIPTION
**Pull request by VladFomenko**

### CODE REVIEW

File: app/controllers/api/v1/cart_items_controller.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  def create
+    result = find_or_create_uniq_record(@cart_item, cart_items_params)
+
+    render_success(data: result, status: 201)
+  end
+
+  def update
+    @cart_item.update(cart_items_params)
+
+    render_success data:@cart_item
```
Comment (oleksiimikhno) [inline app/controllers/api/v1/cart_items_controller.rb:20]: You wrote a different style of method render_success with () or without

File: app/controllers/api/v1/cart_items_controller.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
+class Api::V1::CartItemsController < ApplicationController
+  include CartItemableController
+  before_action :cart_items_params, :set_cart, :set_product, :set_cart_item, except: :show
+
+  def show
```
[the remark this answers is not in the document]
Comment (VladFomenko) [reply app/controllers/api/v1/cart_items_controller.rb:5]: I thought about it and if I want to see only one CartItem I need this, but if I have CartItem controller and show actions is a standard methods CRUD I think it's ok

File: app/controllers/api/v1/cart_items_controller.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    cart_item = CartItem.find(params[:id])
+
+    render_success data: cart_item
+  end
+
+  def create
+    result = find_or_create_uniq_record(@cart_item, cart_items_params)
+
+    render_success(data: result, status: 201)
```
[the remark this answers is not in the document]
Comment (VladFomenko) [reply app/controllers/api/v1/cart_items_controller.rb:14]: agree

File: config/routes.rb [hunk 4]
```diff
@@ -15,9 +15,9 @@
       post '/login', to: 'sessions#create'
       post '/password_update', to: 'passwords#update'
 
-      resources :users
-      resources :products
-      resources :carts
+      resources :users, :products, :carts
```
[the remark this answers is not in the document]
Comment (VladFomenko) [reply config/routes.rb:18]: okay, Can use :except, less code

File: db/seeds.rb [hunk 5]
```diff
@@ -1,13 +1,24 @@
 require 'ffaker'
 
+STUB = 'Just the text to fill in the required number of characters. Just the text to fill in the required number of characters.'
```
[the remark this answers is not in the document]
Comment (VladFomenko) [reply db/seeds.rb:3]: Good point, thanks!

File: spec/factories/cart_item.rb [hunk 6]
```diff
@@ -0,0 +1,8 @@
+FactoryBot.define do
+  factory :cart_item do
+    quantity { 100 }
+    price { 1000.0 }
+    association :cart
```
[the remark this answers is not in the document]
Comment (VladFomenko) [reply spec/factories/cart_item.rb:5]: I think it's the same

File: spec/requests/api/v1/cart_items_spec.rb [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+  end
+
+  path '/api/v1/cart_items/{id}' do
+    let(:id) { cart_item.id }
+
+    parameter name: :cart_id, in: :query, type: :integer, description: 'ID of the cart'
+    parameter name: :product_id, in: :query, type: :integer, description: 'ID of the product'
+    parameter name: :id, in: :path, type: :integer, description: 'ID of the cart_item'
+
+    get('show cart_item') do
```
[the remark this answers is not in the document]
Comment (VladFomenko) [reply spec/requests/api/v1/cart_items_spec.rb:48]: I want to receive all cart item with serializer.

File: spec/factories/cart_item.rb (hunk 6 above)
[the remark this answers is not in the document]
Comment (oleksiimikhno) [reply spec/factories/cart_item.rb:5]: In this code association is nil, when you add , factory: : category. That created category with association to product

**Pull request merged**